src/OrchardCore/Orchard.Localization.Abstractions/PluralArgument.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Orchard.Localization.Abstractions
+{
+    public class PluralArgument
```

Comment:
I thought it won't be a problem to remove it, but it simplifies pluralization in `StringLocalizer`. It unequivocally identifies pluralization data in the argument list.

Better approach would be to have a separate indexer in the localizer that accepts 'count' and 'plural forms'

`LocalizedString this[int count,  string[] pluralForms, params object[] arguments]`

but we are unable to extent `IStringLocalizer` interface. If we define our own interface we will also need to provide our own implementation for `IHtmlLocalizer` and `IViewLocalizer` ... we would reimplement big part of ASP.NET Localization. 

If we pass pluralization data as an argument to `IStringLocalizer` we can provide only our implementation of `IStringLocalizer` and handle pluralization there. `PluralArgument` is created by `IStringLocalizer.Plural` extension method and it is passed as the last argument to `IStringLocalizer`.

`T.Plural(1, new [] {"goal", "goals"}, someOtherArgument)`

translates to

`T["goal", someOtherArgument, new PluralArgument() { Count = 1, PluralForms =new [] {"goal", "goals"}}`